src/app/page.tsx

```diff
@@ -1,13 +1,102 @@
+"use client";
 import Image from "next/image";
+import { useEffect, useState } from "react";
+import { abi } from "./utils/abi";
+import { useSearchParams } from "next/navigation";
+import { ConnectButton } from "@rainbow-me/rainbowkit";
+import {
+  useAccount,
+  useWriteContract,
+  useWaitForTransactionReceipt,
+} from "wagmi";
+import { checkAndRequestSFuel } from "./utils/sfuelClaim";
+import { SyncLoader } from "react-spinners";
+import toast, { Toaster } from "react-hot-toast";
+import { performMulticall } from "./multicall";
+import { MIN_SFUEL_BALANCE, CONTRACT_ADDRESS } from "./utils/constants";
 
 export default function Home() {
+  const [isClaiming, setIsClaiming] = useState(false);
+  const { address, isConnected } = useAccount();
+  const searchParams = useSearchParams();
+
+  const searchHash =
+    searchParams.get("hash") ||
+    "0x0000000000000000000000000000000000000000000000000000000000000000";
+
+  const {
+    data: hash,
+    isPending,
+    isError,
+    error,
+    writeContract,
+  } = useWriteContract();
+  const { isSuccess, isLoading } = useWaitForTransactionReceipt({
+    hash: hash,
+  });
+
+  useEffect(() => {
+    if (isSuccess) {
+      toast.success("Claimed Cards Successfully");
+    }
+  }, [isSuccess]);
+
+  useEffect(() => {
+    if (isError) {
+      toast.error("Error during claim process");
+      console.error("Error during claim process:", error);
+    }
+  }, [isError, error]);
+
+  useEffect(() => {
+    performMulticall().catch(console.error);
+  }, []);
+
+  const onClaim = async () => {
```

Comment:
onclick should also check for if the correct network is connected, right now it gives me a txn on whatever chain I am connected on.